_posts/2022-02-17-quarkus-superheroes-to-the-rescue.adoc

```diff
@@ -0,0 +1,239 @@
+
+An additional user interface, shown in Figure 2, displays previous fight statistics, including the percentage of wins for heroes vs. villains and the top 10 winners and the number of wins for each.
+
+.Fight Statistics
+image::event_stats.png[alt=Event Statistics UI]
+
+== Under the Cape!
+
+Figure 3 shows the overall architecture of the application.
+
+.Overall Architecture
+image::application_architecture.png[alt=Overall architecture]
+
+There are four Quarkus applications, each of which will be described further below, an http://angular.io/[Angular^] application, and some backing services: https://prometheus.io[Prometheus^], https://www.postgresql.org[PostgreSQL^], https://kafka.apache.org[Apache Kafka^], and https://www.apicur.io/registry[Apicurio Schema Registry^].
+
+=== Hero Service
+
+The Hero service is a fully reactive and non-blocking HTTP microservice exposing CRUD operations on a `+Hero+` data model. The REST layer is implemented using https://quarkus.io/guides/resteasy-reactive[RESTEasy Reactive^] with reactive endpoints. Data is persisted into a https://www.postgresql.org[PostgreSQL^] database using the https://quarkus.io/guides/hibernate-reactive-panache#solution-2-using-the-repository-pattern[repository pattern^] from https://quarkus.io/guides/hibernate-reactive-panache[Quarkus Hibernate Reactive with Panache^]. Additionally, the Hero service favors https://quarkus.io/guides/cdi-reference#simplified-constructor-injection[constructor injection^] of beans over https://quarkus.io/guides/cdi-reference#qualified-injected-fields[field injection^]. The Hero service contains a full suite of unit and integration tests, using https://quarkus.io/guides/getting-started-testing#further-simplification-with-injectmock[mocking^], https://quarkus.io/guides/getting-started-testing#using-spies-instead-of-mocks-with-injectspy[spying^], and testing within https://quarkus.io/guides/hibernate-reactive-panache#transactions[isolated transactions^].
+
+There are four https://quay.io/repository/quarkus-super-heroes/rest-heroes?tab=tags[container images available for the Hero service^]: `+java11-latest+`, `+java17-latest+`, `+native-java11-latest+`, and `+native-java17-latest+`.
+
+[NOTE]
+====
+For performance reasons, only 100 random heroes from the https://github.com/quarkusio/quarkus-super-heroes/blob/characterdata/all-heroes.sql[3,458 total available heroes^] are loaded into the database. If you'd like to load more data, grab the https://github.com/quarkusio/quarkus-super-heroes/blob/characterdata/all-heroes.sql[`+all-heroes.sql+` from the `+characterdata+` branch in the repo^]. Copy the data into https://github.com/quarkusio/quarkus-super-heroes/blob/main/rest-heroes/src/main/resources/import.sql[`+rest-heroes/src/main/resources/import.sql+`^] as well as into https://github.com/quarkusio/quarkus-super-heroes/blob/main/rest-heroes/deploy/db-init/initialize-tables.sql[`+rest-heroes/deploy/db-init/initialize-tables.sql+`^].
```

Comment:
When I read performance reasons I assume there is something inherent bad :)

I assume what you mean more specifically is that "To have fast startup during development, only 100 ..." as I assume the system doesn't exactly run slow just because you load more data ?

Reply:
You are right. I will change it. Good catch.

Reply:
Fixed